cypress-tests/cypress/support/redirectionHandler.js

```diff
@@ -395,6 +395,15 @@ function upiRedirection(
 }
 
 function verifyReturnUrl(redirection_url, expected_url, forward_flow) {
+  const urlParams = new URLSearchParams(redirection_url.search);
+  const paymentStatus = urlParams.get('status');
+
+  // Check for valid statuses
+  if (paymentStatus !== 'succeeded' && paymentStatus !== 'processing') {
+    throw new Error(`Test failed: Unexpected payment status: ${paymentStatus}`);
```

Comment:
or better, we can have a more contextual error something as below:
```suggestion
    throw new Error(`Payment failed after redirection with status: ${paymentStatus}`);
```